state/machine.go

```diff
@@ -601,11 +601,12 @@ func (m *Machine) forceDestroyOps(maxWait time.Duration) ([]txn.Op, error) {
 		// We set the machine to Dying if it isn't already dead.
 		var machineOp txn.Op
 		if m.Life() < Dead {
-			// Make sure we don't want the vote, and we are queued to be Dying
+			// Make sure we don't want the vote, and we are queued to be Dying.
+			// Since we are force deleting, life assert should be current machine's life.
 			machineOp = txn.Op{
 				C:      machinesC,
 				Id:     m.doc.DocID,
-				Assert: bson.D{{"life", bson.D{{"$in", []Life{Alive, Dying}}}}},
+				Assert: bson.D{{"life", m.doc.Life}},
```

Comment:
The original assertion is actually *more* permissive than this one. I'd be hesitant to tighten it up unless we're 1000% sure that it's never an expected outcome that the life could go from alive to dying as this transaction is being composed.

Reply:
I am not sure why you think it is more permissive? Previously it was 'either alive or dying', now it is 'whatever is current live' :)

Reply:
Before, the current life could be "alive", but if the life were to change to "dying" by the time the txn was run, it would still satisfy the assert. With the change in this PR, that is no longer the case.

Reply:
I see.
Well, the life is kind of irrelevant during force... Maybe it should not be part of the assert at all...
Meanwhile I've reverted this change.